### PR TITLE
User upload bug fix

### PR DESCRIPTION
**Pull request by Simpcyclassy**

### CODE REVIEW

File: components/home/reducers.js [hunk 1]
```diff
@@ -1,34 +1,41 @@
...
-        userUpload.reactions[type] += 1;
-    } else {
-        userUpload.reactions[type] -= 1;
-    }
-    updatedUploads.set(id, userUpload);
+    const upload = updatedUploads.get(id);
+
+    uploads.forEach(() => {
+        const { users: { userId } } = upload.reactions[type];
+        if (!userId) {
```
Comment (enyeInc) [inline components/home/reducers.js]: this doesn't make any sense Chioma, why are you checking for the `userId` key in users?

File: models/fakeUploads.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            maybe: {
+                count: 0,
+                users: {},
+            },
+        },
+        user: {
+            image: faker.image.avatar(),
+            name: faker.name.findName(),
+            userId: uuid(),
+        },
```
Comment (enyeInc) [inline models/fakeUploads.js:28]: what is the reasoning behind attaching a `user` object to each upload object?

File: models/fakeUsers.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    },
+    {
+        id: uuid(),
+    },
+    {
+        id: uuid(),
+    },
+];
+
+export default fakeUsers;
```
Comment (enyeInc) [inline models/fakeUsers.js]: what are you using this file for?

**Comment by enyeInc** [comment]
@Simpcyclassy Did you test this code to make sure that its working?

File: models/fakeUploads.js (hunk 2 above)
Comment (Simpcyclassy) [reply models/fakeUploads.js:28]: I created a separate file for it later. It was a pattern I was supposed to reverse.

File: models/fakeUsers.js (hunk 3 above)
Comment (Simpcyclassy) [reply models/fakeUsers.js]: To generate user Ids

File: components/home/reducers.js (hunk 1 above)
Comment (Simpcyclassy) [reply components/home/reducers.js]: The object is supposed to save user Ids that have reacted to a type of reaction, so the idea is to check if a particular user Id exists in that object.

File: models/fakeUsers.js (hunk 3 above)
Comment (enyeInc) [reply models/fakeUsers.js]: But where are you using it? Also there is no need for you to generate userIds.

File: components/home/reducers.js (hunk 1 above)
Comment (enyeInc) [reply components/home/reducers.js]: Are you checking for a particular user id or a key called `userId`? Those are not the same thing

File: models/fakeUploads.js (hunk 2 above)
Comment (enyeInc) [reply models/fakeUploads.js:28]: You can user this partner for now, but what it have to be is `uploader` not user. The user is the person that is logged in but not necessary always the person that is the uploader (given that they will be looking at a feed of uploads mainly from their friends). What you should do is add a fake user in your redux initial state. And use that to drive any logic where you need the user `id (not userId), name or image`.

File: components/home/reducers.js [hunk 4]
```diff
@@ -1,34 +1,43 @@
...
-    } else {
-        userUpload.reactions[type] -= 1;
-    }
-    updatedUploads.set(id, userUpload);
+    const upload = updatedUploads.get(id);
+    const userId = uuid();
+
+    uploads.forEach(userUpload => {
+        const { users } = userUpload.reactions[type];
+        if (Object.values(users).includes(userId[userId])) {
```
Comment (enyeInc) [inline components/home/reducers.js]: what is `userId[userId]`?

File: components/home/reducers.js [hunk 5]
```diff
@@ -1,34 +1,43 @@
...
+            delete users.userId;
+            userUpload.reactions[type].count -= 1;
+        } else {
+            users.userId = userId[userId];
+            userUpload.reactions[type].count += 1;
+        }
+    });
```
Comment (enyeInc) [inline components/home/reducers.js:40]: Does this code work?

File: models/fakeUsers.js (hunk 3 above)
Comment (Simpcyclassy) [reply models/fakeUsers.js]: Nowhere for now.

File: components/home/reducers.js (hunk 4 above)
Comment (Simpcyclassy) [reply components/home/reducers.js]: It is supposed to be the value of UserId.

File: components/home/reducers.js (hunk 5 above)
Comment (Simpcyclassy) [reply components/home/reducers.js:40]: It does work.

File: models/fakeUsers.js (hunk 3 above)
Comment (enyeInc) [reply models/fakeUsers.js]: delete this file

File: components/home/constants.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ export const NAME = 'home';
 
 /** @constant */
 export const INITIAL_STATE = {
+    fakeUserId: ' ',
```
Comment (enyeInc) [inline components/home/constants.js]: delete this, given that you are using the userId in `reducer.js`

File: models/fakeUsers.js (hunk 3 above)
Comment (Simpcyclassy) [reply models/fakeUsers.js]: OK

File: components/home/constants.js (hunk 6 above)
Comment (Simpcyclassy) [reply components/home/constants.js]: Alright.

**Review by enyeInc — APPROVED** [review]
LGTM

**Pull request merged**